packages/extension-dapp/src/index.ts

```diff
@@ -184,7 +185,11 @@ export async function web3FromAddress (address: string): Promise<InjectedExtensi
   }
 
   const accounts = await web3Accounts();
-  const found = address && accounts.find((account): boolean => account.address === address);
+  let found: InjectedAccountWithMeta | undefined;
+
+  if (address) {
+    found = accounts.find((account): boolean => u8aEq(decodeAddress(account.address), decodeAddress(address)));
```

Comment:
```suggestion
    const accountU8a = decodeAddress(address);
    
    found = accounts.find((account): boolean => u8aEq(decodeAddress(account.address), accountU8a));
```

The reason for the suggestion is that we don't need to decode this on each iteration. (We could use `addressEq` but opt for the lower-level version so we don't need to do the same work each iteration until found)